src/rust/zh.manhuaren/src/helper.rs

```diff
@@ -0,0 +1,151 @@
+use aidoku::{
+	error::Result,
+	prelude::*,
+	std::String,
+	std::{
+		net::{HttpMethod, Request},
+		Vec,
+	},
+};
+use md5::{Digest, Md5};
+
+pub fn encode_uri_component(string: String) -> String {
+	let mut result: Vec<u8> = Vec::with_capacity(string.len() * 3);
+	let hex = "0123456789ABCDEF".as_bytes();
+	let bytes = string.as_bytes();
+
+	for byte in bytes {
+		let curr = *byte;
+		if curr.is_ascii_alphanumeric() || curr == 45 || curr == 95 {
+			// 45: -
+			// 95: _
+			result.push(curr);
+		} else {
+			result.push(b'%');
+			result.push(hex[curr as usize >> 4]);
+			result.push(hex[curr as usize & 15]);
+		}
+	}
+
+	String::from_utf8(result).unwrap_or_default()
+}
+
+pub fn encode_uri(string: String) -> String {
```

Comment:
is there a reason you can't use `encode_uri` from aidoku-rs helpers crate?
```toml
[dependencies]
aidoku = { git = "https://github.com/Aidoku/aidoku-rs", features = ["helpers"] }
```
then
```rs
use aidoku::helpers::uri::encode_uri_component;
```